builder/frameworks/espidf.py

```
@@ -244,48 +307,323 @@ def extract_flag_name(line):
+                pass
+        if not f_boot:
+            f_boot = board.get("build.f_boot", None)
+        
+        # Determine the frequencies to use
+        esptool_flash_freq = f_flash  # Always use f_flash for esptool compatibility
+        compile_freq = f_boot if f_boot else f_flash  # Use f_boot for compile-time if available
+        
+        if f_flash and compile_freq:
+            # Ensure frequency compatibility (>= 80MHz must be identical for Flash and PSRAM)
+            compile_freq_val = int(str(compile_freq).replace("000000L", ""))
+            esptool_freq_val = int(str(esptool_flash_freq).replace("000000L", ""))
```

Ruff (0.13.3)

Local variable `esptool_freq_val` is assigned to but never used

Remove assignment to unused variable `esptool_freq_val`

(F841)